node/src/command.rs

```diff
@@ -19,71 +37,64 @@ use std::{io::Write, net::SocketAddr};
+	fn is_manta(&self) -> bool {
+		<dyn sc_service::ChainSpec>::is_manta(self)
+	}
+	fn is_calamari(&self) -> bool {
+		<dyn sc_service::ChainSpec>::is_calamari(self)
+	}
+}
+
 fn load_spec(
 	id: &str,
 	para_id: ParaId,
 ) -> std::result::Result<Box<dyn sc_service::ChainSpec>, String> {
 	Ok(match id {
-		// manta-pc chainspec
-		#[cfg(feature = "manta-pc")]
-		"manta-pc-dev" => Box::new(chain_specs::manta_pc_development_config(para_id)),
-		#[cfg(feature = "manta-pc")]
-		"manta-pc-local" => Box::new(chain_specs::manta_pc_local_config(para_id)),
-		#[cfg(feature = "manta-pc")]
-		"manta-pc-testnet" => Box::new(chain_specs::manta_pc_testnet_config(para_id)),
+		// manta chainspec
+		"manta-dev" => Box::new(chain_specs::manta_development_config(para_id)),
+		"manta-local" => Box::new(chain_specs::manta_local_config(para_id)),
+		"manta-testnet" => Box::new(chain_specs::manta_testnet_config(para_id)),
 		// calamari chainspec
-		#[cfg(feature = "calamari")]
 		"calamari-dev" => Box::new(chain_specs::calamari_development_config(para_id)),
-		#[cfg(feature = "calamari")]
 		"calamari-local" => Box::new(chain_specs::calamari_local_config(para_id)),
-		#[cfg(feature = "calamari")]
 		"calamari-testnet" => Box::new(chain_specs::calamari_testnet_config(para_id)?),
-		#[cfg(feature = "calamari")]
 		"calamari-testnet-ci" => Box::new(chain_specs::calamari_testnet_ci_config()?),
-		#[cfg(feature = "calamari")]
 		"calamari" | "" => Box::new(chain_specs::calamari_config()?),
 		path => {
-			let path = std::path::PathBuf::from(path);
-
-			let starts_with = |prefix: &str| {
-				path.file_name()
-					.map(|f| f.to_str().map(|s| s.starts_with(&prefix)))
-					.flatten()
-					.unwrap_or(false)
-			};
+			let chain_spec = chain_specs::ChainSpec::from_json_file(path.into())?;
 
-			if starts_with("manta-pc") {
-				#[cfg(feature = "manta-pc")]
-				{
-					Box::new(chain_specs::MantaPCChainSpec::from_json_file(path)?)
-				}
-				#[cfg(not(feature = "manta-pc"))]
-				panic!("manta-pc runtime is not available.")
-			} else if starts_with("calamari") {
-				#[cfg(feature = "calamari")]
-				{
-					Box::new(chain_specs::CalamariChainSpec::from_json_file(path)?)
-				}
-				#[cfg(not(feature = "calamari"))]
-				panic!("calamari runtime is not available.")
+			if chain_spec.is_manta() {
+				Box::new(chain_specs::MantaChainSpec::from_json_file(path.into())?)
+			} else if chain_spec.is_calamari() {
+				Box::new(chain_specs::CalamariChainSpec::from_json_file(path.into())?)
 			} else {
-				panic!("Please input a file name starting with manta-pc or calamari.")
+				panic!("Please input a file name starting with manta or calamari.")
 			}
 		}
 	})
 }
 
 impl SubstrateCli for Cli {
 	fn impl_name() -> String {
-		cfg_if::cfg_if! {
-			if #[cfg(feature = "manta-pc")] {
-				"Manta Parachain Collator".into()
-			} else {
-				"Calamari Parachain Collator".into()
-			}
-		}
+		"Manta/Calamari Collator".into()
```

Comment:
We cannot change this name at will. It will affect whether polkadot-js can find our logo.
Just take a look at this place: https://github.com/polkadot-js/apps/blob/master/packages/apps-config/src/ui/logos/index.ts#L308

Reply:
We should add a comment about this.

Reply:
Yeah, I will put a switch here.

Reply:
I think there is a way to do this without affecting polkadot.js, you can see statemine's info does not match its `rpc.system.name()`.

It is not possible to add a switch here since it may not have a spec loaded yet.
 
<img width="1625" alt="Screen Shot 2021-11-04 at 11 39 07 AM" src="https://user-images.githubusercontent.com/720571/140400156-f42b65b3-8a19-4508-8a75-02ab7ed7c596.png">

Reply:
So, there must be a way to find the logo respectively without depending on by `impl_name`. I found kusama and polkadot have the same `impl_name`, but polkadot-js shows the correct logo.

Anyway, we have to submit a pr to fix it if we're going to use this `impl_name`.

Reply:
Perhaps `Manta Collator` is enough. I don't know, but acala is using `Acala Node`.

Reply:
`rpc.system.chain()` works well enough. PR already submitted: https://github.com/polkadot-js/apps/pull/6493